xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon_base.h

```diff
@@ -249,24 +249,33 @@ extern "C"
     bool (*is_setting_using_default)(void* kodiBase, const char* id);
   } AddonToKodiFuncTable_Addon;
 
+  typedef ADDON_STATUS(ATTR_APIENTRYP PFN_KODI_ADDON_CREATE_V1)(const KODI_HANDLE first_instance,
+                                                                KODI_ADDON_HDL* hdl);
+  typedef void(ATTR_APIENTRYP PFN_KODI_ADDON_DESTROY_V1)(const KODI_ADDON_HDL hdl);
+  typedef ADDON_STATUS(ATTR_APIENTRYP PFN_KODI_ADDON_CREATE_INSTANCE_V1)(const KODI_ADDON_HDL hdl,
+                                                                         int instanceType,
+                                                                         const char* instanceID,
+                                                                         KODI_HANDLE instance,
+                                                                         const char* version,
+                                                                         KODI_HANDLE* addonInstance,
+                                                                         KODI_HANDLE parent);
+  typedef void(ATTR_APIENTRYP PFN_KODI_ADDON_DESTROY_INSTANCE_V1)(const KODI_ADDON_HDL hdl,
+                                                                  int instanceType,
+                                                                  KODI_HANDLE instance);
+  typedef ADDON_STATUS(ATTR_APIENTRYP PFN_KODI_ADDON_SET_SETTING_V1)(const KODI_ADDON_HDL hdl,
+                                                                     const char* settingName,
+                                                                     const void* settingValue);
```

Comment:
The add about the typedef's is the most needed part within my sandbox, as this becomes several times used on the autogeneration to create the library for it.

The reason why these are only for calls from Kodi to add-ons has the background that the add-on has to set its pointers there.

In the case of callbacks from add-ons to Kodi, it will be stored in the sandbox as correct functions so that it can be used very easily and cleanly in "C" or as a basis for other languages, see https://github.com/AlwinEsch/kodi/blob/new-interface-2/xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon_base.h#L183-L249.